hscontrol/derp/server/derp_server.go

```diff
@@ -360,3 +371,14 @@ func serverSTUNListener(ctx context.Context, packetConn *net.UDPConn) {
 		}
 	}
 }
+
+type HeadscaleTransport struct {
+	verifyHandler http.HandlerFunc
+}
+
+func (t *HeadscaleTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	recorder := httptest.NewRecorder()
+	t.verifyHandler(recorder, req)
+	resp := recorder.Result()
```

Comment:
I think instead of doing the recorder here, we can just call the underlying logic of the verify handler and construct a Response, if it is not a seperate function, I think the logic should be factored out so you can call `verifyNode` (or something), and that is used here and in the `verifyHandler`.